### PR TITLE
Customizable layouts

### PR DESCRIPTION
**Pull request by alexkuz**
Phew!
So, I refactored layouts a little bit. Now user can provide JSON with layout configurations, which is then converted to internal layout object. Layout configuration describes panels (or columns), each contains views. Each panel and view has `position` object, that have a `size` or `grow` property. `size` defines exact width for panels and height for views, and `grow` works like CSS `flex-grow`. So panel with `{ grow: 2 }` will be twice as wide as panel with `{ grow: 1 }`.

There are also some fixes to prevent crashes, `view.layoutConfig` is now optional (view is hidden if there is no config).

### CODE REVIEW

**Comment by jasonwilson** [comment]
@alexkuz it looks like CI is failing with these changes. Would you mind adding/fixing tests for this functionality?

File: lib/dashboard.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ var EventLoopView = require("./views/eventloop-view");
 var MemoryView = require("./views/memory-view");
 var CpuView = require("./views/cpu-view");
 var HelpView = require("./views/help");
-var layouts = require("./layouts");
+var generateLayouts = require("./generate-layouts");
```
Comment (jasonwilson) [inline lib/dashboard.js]: This file doesn't seem to exist?

Comment (alexkuz) [inline lib/dashboard.js]: Oops! Missed it.

**Comment by alexkuz** [comment]
I'll fix tests, you can try it out meanwhile (with `-l ./test/app/layouts.json`, for example)

Forgot to mention - I also removed `--interleave` parameter, since it can be configured now via layouts.

**Comment by alexkuz** [comment]
@jasonwilson tests are ready

With a bit more of refactoring, it would be easy to implement `status` view with this, like:

```js
{
  name: "stdout",
  type: "log",
  streams: ["stdout"],
  exclude: "^\[STATUS\]"
},
{
  name: "stderr",
  type: "log",
  streams: ["stderr"],
  exclude: "^\[STATUS\]"
},
{
  position: {
    size: 3
  },
  name: "status",
  type: "log",
  streams: ["stdout", "stderr"],
  include: "^\[STATUS\](.*)"
}
```

File: lib/generate-layouts.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+var fs = require("fs");
+/* eslint-disable no-magic-numbers */
+
+// Each layout consists of vertical panels, that contains its position and horizontal views.
+// Flex-like positions of panels and views defined by 'grow' and 'size' parameters.
+// View or panel with 'size' has exactly <size> height or width respectively.
+// View or panel with 'grow' fills <grow> part of the residuary space (it works like flex-grow).
+// By default, position = { grow: 1 }
+
+var DEFAULT_LAYOUT_CONFIG = [
```
Comment (alexkuz) [inline lib/generate-layouts.js]: You are right, js is better here. I replaced `fs` with `require` - not completely sure if I require there in a right way, but it seems to work.

**Comment by alexkuz** [comment]
Added layout config validation with `jsonschema` (it seems to be the most compact JSON-Schema lib)

**Pull request merged**